docs/user-guide/KubernetesCurator.rst

```diff
@@ -0,0 +1,386 @@
+Some examples below assume you have this Pod running to copy to and from the PVC. 
+
+.. code-block:: bash
+
+    PVC_NAME=nemo-workspace
+    MOUNT_PATH=/nemo-workspace
+
+    kubectl create -f - <<EOF
+    apiVersion: v1
+    kind: Pod
+    metadata:
+      name: nemo-workspace-busybox
+    spec:
+      containers:
+      - name: busybox
+        image: busybox
+        command: ["sleep", "infinity"]
+        volumeMounts:
+        - name: workspace
+          mountPath: ${MOUNT_PATH}
+      volumes:
+      - name: workspace
+        persistentVolumeClaim:
+          claimName: ${PVC_NAME}
+    EOF
+
+Feel free to delete this container if no longer needed, but it should use very little
+resources when idle.
+
+.. code-block:: bash
+
+    kubectl delete pod nemo-workspace-busybox
+
+Setup Docker Secrets
+--------------------
+A Kubernetes Secret needs to be created on the k8s cluster to authenticate with the NGC
+private registry. If not done already, get an NGC key from ngc.nvidia.com. Create a
+secret key on the k8s cluster with (replace ``<NGC KEY HERE>`` with your NGC secret
+key. Note that if you have any special characters in your key you might need to wrap
+the key in single quotes (``'``) so it can be parsed correctly by k8s)::
+
+    kubectl create secret docker-registry ngc-registry --docker-server=nvcr.io --docker-username=\$oauthtoken --docker-password=<NGC KEY HERE>
+
+Setup Python Environment
+------------------------
+
+The environment to run the provided scripts in this example does not need the full
+``nemo_curator`` package, so you can create a virtual environment with just the
+required packages as follows:
+
+.. code-block:: bash
+
+    python3 -m venv venv
+    source venv/bin/activate
+
+    pip install 'dask_kubernetes>=2024.4.1'
+
+Upload Data to PVC
+------------------
+
+To copy into the ``nemo-workspace`` PVC, we will do so with ``kubectl exec``. You may also
+use ``kubectl cp``, but ``exec`` has fewer surprises regarding compressed files:
+
+.. code-block:: bash
+  
+    # Replace <...> with a path on your local machine 
+    LOCAL_WORKSPACE=<...>
+
+    # This copies $LOCAL_WORKSPACE/my_dataset to /my_dataset within the PVC.
+    # Change foobar to the directory or file you wish to upload.
+    ( cd $LOCAL_WORKSPACE; tar cf - my_dataset | kubectl exec -i nemo-workspace-busybox -- tar xf - -C /nemo-workspace )
+
+.. note::
+    See :ref:`data-curator-download` for an example of how to download local data that can be uploaded to the PVC
+    with the above instruction.
+
+Create a Dask Cluster
+---------------------
+
+Use the ``create_dask_cluster.py`` to create a CPU or GPU dask cluster.
+
+.. note::
+    If you are creating another Dask cluster with the same ``--name <name>``, first delete it via::
+
+        kubectl delete daskcluster <name>
+
+.. code-block:: bash
+
+    # Creates a CPU Dask cluster with 1 worker
+    python create_dask_cluster.py \
```

Comment:
This command appears to have a lot of overlap with the built-in `dask kubernetes gen cluster` command. I wonder if things can be made more flexible there so it can be used directly instead?

Reply:
Cool, I wasn't aware of that. It looks like the flags are similar to what `make_cluster_spec` accepts
```sh
dask kubernetes gen cluster --help
Usage: dask kubernetes gen cluster [OPTIONS]

  Generate a DaskCluster spec, see ``dask_kubernetes.operator.make_cluster_spec``.

Options:
  --name TEXT                    Name of the DaskCluster  [required]
  --image TEXT                   Container image to use (default
                                 'ghcr.io/dask/dask:latest')
  --n-workers INTEGER            Number of workers
  --resources TEXT               Resource constraints
  -e, --env TEXT                 Environment variables
  --worker-command TEXT          Worker command (default 'dask-worker')
  --scheduler-service-type TEXT  Service type for scheduler (default 'ClusterIP')
  --jupyter                      Start Jupyter on the scheduler (default 
```

So someone still needs to edit the spec to, say, add PVC mounts or remove `resources` from the scheduler (e.g., GPUs or IB devices), they'd need a script like this to do that. In light of this, do you still see a path of using `dask kubernetes gen cluster` by adding some of these configuration options, or do you think this opinionated script still has value?

Reply:
I was thinking we might be able to satisfy most of what you need by adding a couple more flags. 

If all you need is a flag to set pvc mounts and a flag to set scheduler resources independently then I could see us adding that quite easily.